app/gui2/src/util/config.ts

```diff
@@ -0,0 +1,477 @@
+      groups[name] = group
+      group.name = name
+    }
+    group.setPath([name])
+  }
+
+  /** Set the path of this group. If this group was placed in another group, the path will contain
+   * all the parent group names. */
+  setPath(path: string[]) {
+    for (const option of Object.values(this.options)) {
+      option.path = path
+    }
+    for (const [name, group] of Object.entries(this.groups)) {
+      group.setPath(path.concat([name]))
+    }
+  }
+
+  /** Return a deep copy of this group definition. */
+  clone(): this {
+    const result: AnyGroup = new Group()
+    for (const [name, group] of Object.entries(this.groups)) {
+      result.groups[name] = group.clone()
+    }
+    for (const [name, option] of Object.entries(this.options)) {
+      result.options[name] = option.clone()
+    }
+    return result as this
+  }
+
+  /** Merge this group definition with another group definition. Returns a deeply merged group. In
+   * case the argument will override some options, errors will be logged. */
+  merge<Other extends AnyGroup>(other?: Other | null): this & Other {
+    if (other == null) {
+      return this as this & Other
+    } else {
+      const result: AnyGroup = new Group()
+      Object.assign(result.groups, this.groups)
+      for (const [otherGroupName, otherGroup] of Object.entries(other.groups)) {
+        const group = result.groups[otherGroupName]
+        if (group == null) {
+          result.groups[otherGroupName] = otherGroup
+        } else {
+          result.groups[otherGroupName] = group.merge(otherGroup)
+        }
+      }
+      Object.assign(result.options, this.options)
+      for (const [otherOptionName, otherOption] of Object.entries(other.options)) {
+        const option = result.options[otherOptionName]
+        if (option != null) {
+          console.error(`Duplicate config option found '${option.qualifiedName()}'.`)
+        }
+        result.options[otherOptionName] = otherOption
+      }
+      result.name = this.name
+      result.description = this.description
+      return result as this & Other
+    }
+  }
+
+  load(config: StringConfig, stack: string[] = []): string[] {
+    let unrecognized: string[] = []
+    const addUnrecognized = (name: string) => {
+      unrecognized.push(stack.concat([name]).join('.'))
+    }
+    for (const [key, value] of Object.entries(config)) {
+      if (typeof value === 'string') {
+        const option = this.options[key]
+        if (option == null) {
+          addUnrecognized(key)
+        } else {
+          option.load(value)
+        }
+      } else {
+        const group = this.groups[key]
+        if (group == null) {
+          addUnrecognized(key)
+        } else {
+          const subStack = stack.concat([key])
+          unrecognized = unrecognized.concat(group.load(value, subStack))
+        }
+      }
+    }
+    return unrecognized
+  }
+
+  loadAllAndDisplayHelpIfUnsuccessful(configs: (StringConfig | null | undefined)[]): boolean {
```

Comment:
Loading (validation) and displaying an error should be two separate activities. I believe this should simply go away once the help page is in vue.